src/WebJobs.Script.WebHost/Diagnostics/MetricsEventManager.cs

```diff
@@ -501,6 +511,16 @@ private void RaiseFunctionMetricEvent(RunningFunctionInfo runningFunctionInfo, i
                         currentTime,
                         runningFunctionInfo.StartTime);
                 }
+
+                _meshInitServiceClient?.PublishContainerFunctionExecutionActivity(
+                    new ContainerFunctionExecutionActivity()
+                    {
+                        EventTime = DateTime.UtcNow,
+                        FunctionName = runningFunctionInfo.Name,
+                        ExecutionStage = runningFunctionInfo.ExecutionStage,
+                        TriggerType = runningFunctionInfo.TriggerType,
+                        Success = runningFunctionInfo.Success
+                    });
```

Comment:
Would it make sense to batch and subsequently drain these rather than doing a publish every single time? I'm assuming that would consequently impact scaling decisions as well, right ?

Reply:
I don't think we can have much lag here before publishing the status since it will delay scaleout for the function app as well under load.

Reply:
How does it compare to the overhead of publishing every single time? I was thinking it a balance can be achieved by having periodic flushes like MetricsEventManager does for the some of the other events (_functionActivityFlushIntervalSeconds) .

---
In reply to: [364032310](https://github.com/Azure/azure-functions-host/pull/5432#discussion_r364032310) [](ancestors = 364032310)

Reply:
@mathewc also had similar comment offline. Added another service to aggregate and push in 5 sec intervals.